DEV.md

```diff
@@ -14,54 +18,108 @@ npm run build
 
 `$ npm install -g npm`
 
+Clone the repo:
+
+`$ git clone https://github.com/mozilla/shield-studies-addon-template.git`
+
 After cloning the repo, you can run the following commands from the top level directory, one after another:
 
 ```
 $ npm install
 $ npm run build
 ```
+This packages the add-on into an xpi file which is stored in `dist/`. This file is what you load into Firefox.
+
+## General notes on Shield Study Engineering
+
+Shield study add-ons are legacy (`addon/bootstrap.js`) add-ons with an optional embedded web extension (`addon/webextension/background.js`).
+
+The web extension needs to be packaged together with a legacy add-on in order to be able to access Telemetry data, user preferences etc that are required for collecting relevant data for [Shield Studies](https://wiki.mozilla.org/Firefox/Shield/Shield_Studies).
+
+It is recommended to build necessary logic and user interface using in the context of the webextension and communicate with the legacy add-on code through messaging whenever privileged access is required.
+
+## Loading the Web Extension in Firefox
+
+You can have Firefox automatically launched and the add-on installed by running:
+
+`$ npm run firefox`
+
+To load the extension manually instead, open (preferably) the [Developer Edition of Firefox](https://www.mozilla.org/firefox/developer/) and load the `.xpi` using the following steps:
+
+* Navigate to *about:config* and set `extensions.legacy.enabled` to `true`. This permits the loading of the embedded Web Extension since new versions of Firefox are becoming restricted to pure Web Extensions only.
+* Navigate to *about:debugging* in your URL bar
+* Select "Load Temporary Add-on"
+* Find and select the latest xpi file you just built.
+
+## Seeing the add-on in action
+
+To debug installation and loading of the add-on:
+
+* Navigate to *about:config* and set `shield.testing.logging.level` to `10`. This permits shield-add-on log output in browser console
+* Open the Browser Console using Firefox's top menu at `Tools > Web Developer > Browser Console`. This will display Shield (loading/telemetry) and log output from the add-on.
+
+See [TESTPLAN.md](./TESTPLAN.md) for more details on how to see this add-on in action and hot it is expected to behave.
+
+## Automated launch of Firefox with add-on installed
 
-This packages the add-on into `dist/linked-addon.xpi`. This file is the addon you load into Firefox.
+`$ npm run firefox` starts Firefox and automatically installs the add-on in a new profile and opens the browser console automatically.
 
-Note: `linked-addon.xpi` is a symbolic link to the extension's true XPI, which is named based on the study's unique addon ID specified in `package.json`.
+Note: This runs in a recently created profile. To have the study run despite the eligibility requirement of having at least 1 day old profiles, a config override is set in place to force the study to run.
```

Comment:
What's this note referring to? I'm not aware of any eligibility requirement for profile age. What is the config override pref?